File: Medium/377. Combination Sum IV/377. Combination Sum IV.java
public class Solution {
    public int combinationSum4(int[] nums, int target) {
        int[] dp = new int[target + 1];
        dp[0] = 1; // There's one way to make target 0, which is by not selecting any number.

        for (int i = 1; i <= target; i++) {
            for (int num : nums) {
                if (i >= num) {
                    dp[i] += dp[i - num];
                }
            }
        }

        return dp[target];
    }
}

File: Medium/377. Combination Sum IV/377. Combination Sum IV.rs
impl Solution {
    pub fn combination_sum4(nums: Vec<i32>, target: i32) -> i32 {
        let mut dp = vec![0; (target + 1) as usize];
        dp[0] = 1; // There's one way to make target 0, which is by not selecting any number.
        
        for i in 1..=target {
            for &num in &nums {
                if i >= num {
                    dp[i as usize] += dp[(i - num) as usize];
                }
            }
        }
        
        dp[target as usize]
    }
}
